splunk_connect_for_snmp_poller/manager/poller.py

```diff
@@ -127,22 +130,28 @@ def __check_inventory(self):
                     else:
                         self.update_schedule_for_changed_conf(entry_key, ir, profiles)
 
-            self.clean_job_inventory(inventory_hosts)
+            self.clean_job_inventory(inventory_entry_keys, inventory_hosts)
 
     def delete_all_entries_per_host(self, host):
         for entry_key in list(self._jobs_map.keys()):
             if entry_key.split("#")[0] == host:
                 schedule.cancel_job(self._jobs_map.get(entry_key))
                 del self._jobs_map[entry_key]
 
-    def clean_job_inventory(self, inventory_hosts):
+    def clean_job_inventory(self, inventory_entry_keys: set, inventory_hosts: set):
         for entry_key in list(self._jobs_map):
-            if entry_key not in inventory_hosts:
+            if entry_key not in inventory_entry_keys:
                 logger.debug("Removing job for %s", entry_key)
                 schedule.cancel_job(self._jobs_map.get(entry_key))
                 db_host_id = return_database_id(entry_key)
-                logger.debug("Removing _id %s from mongo database", db_host_id)
-                self._mongo_walked_hosts_coll.delete_host(db_host_id)
+                host_without_ip = db_host_id.split(":")[0]
```

Comment:
shouldn't it be `host_without_port`?

Reply:
yep 😆

Reply:
@weliasz good catch, never used port in inventory so didn;t think about that use case, changed